### ISSUE TITLE
Header check doesn't handle tabs

### ISSUE THREAD

**Issue by danielrcollins1** [checkstyle/checkstyle#12869]
I have read check documentation: https://checkstyle.sourceforge.io/config_header.html#Header
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
D:\xTemp>type CheckStyleConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
    "https://checkstyle.org/dtds/configuration_1_2.dtd">
<module name="Checker">
    <module name="Header">
      <property name="header"
        value="/*\n\tAlice Smith's Program\n*/" />
    </module>
</module>

D:\xTemp>type HelloWorld.java
/*
        Alice Smith's Program
*/
import java.util.*;
public class HelloWorld {
   public static void main(String[] args) {
                System.out.println("Hello, World!");
        }
}

D:\xTemp>set RUN_LOCALE="-Duser.language=en -Duser.country=US"

D:\xTemp>java %RUN_LOCALE% -jar checkstyle-10.9.2-all.jar -c CheckStyleConfig.xml HelloWorld.java
Starting audit...
[ERROR] D:\xTemp\HelloWorld.java:2: Line does not match expected header line of '\tAlice Smith's Program'. [Header]
Audit done.
Checkstyle ends with 1 errors.
```
---

The header check seems unable to handle tabs.
E.g., the Java file in the example above has a lead tab on line 2, and the Header header property specifies a tab in the same place with the \t escape character. However, the check identifies these as non-matching and gives an error. (If the tab indicator is removed from both files, then this example passes the audit without any error.)
It would be nice if the Header check could handle tabs cleanly.

---

**Comment by rnveach** [comment]
I am just pointing out how the check works now, not to do with issue approval.

> `value="/*\n\tAlice Smith's Program\n*/"`

https://checkstyle.org/config_header.html#Header_Properties
The mis-understanding is `header` is pure text and it isn't a regexp or handled as a java string. Special processing has to be done to recognize `\n` as a new line character.
https://github.com/checkstyle/checkstyle/blob/54616f662130aa0ca5afe45f72d36cdefb78a781/src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java#L52-L53

The matching is failing is because it is looking for the text `\` and `t` based on your configuration values and does not see it as a tab which you are looking for.

If you converted the `\t` to an actual tab character, it should work as you want.

**Comment by danielrcollins1** [comment]
That's broadly what I expected. I was on the fence of labeling this as a bug or feature request.

That said: Inserting an actual tab in the config file (which is nontrivial in some editors) fails to work, either. Even inspecting my code & config files in a hex editor to confirm they're aligned, and it doesn't work for me. Take out the tab characters and then it works.

[CheckStyleConfig.xml.txt](https://github.com/checkstyle/checkstyle/files/11010382/CheckStyleConfig.xml.txt)
[HelloWorld.java.txt](https://github.com/checkstyle/checkstyle/files/11010383/HelloWorld.java.txt)

**Comment by rnveach** [comment]
You are correct. It seems the XML parser provided (SAXParserFactory) converts the tab to a single space. You should be able to use `&#009;` as a tab character in the XML document, but this won't be very friendly for people who don't know the conversion. Another option is to switch to `headerFile` to move it outside the configuration file, but some people prefer it all in one file.

**Comment by rnveach** [comment]
@romani ping

**Comment by danielrcollins1** [comment]
Okay, that actually does provide a workaround I can use, thanks for that.

FWIW, my distribution use-case constrains me to having everything in the one config file.

**Comment by romani** [comment]
@danielrcollins1, please share full CLI output  to let others easily reuse your workaround.

**Comment by danielrcollins1** [comment]
```
D:\Teach\!KCC\CP05_2023-03\Assignments\TestSuite\xCheckStyleHeaderTest>type MinimalConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
    "https://checkstyle.org/dtds/configuration_1_2.dtd">
<module name="Checker">
    <module name="Header">
      <property name="header"
        value="/*\n&#009;Name\n&#009;Project\n&#009;CP 500\n*/" />
      <property name="ignoreLines" value="2, 3"/>
    </module>
</module>

D:\Teach\!KCC\CP05_2023-03\Assignments\TestSuite\xCheckStyleHeaderTest>type HelloWorld.java
/*
        Alice Smith
        Hello world program
        CP 500
*/

import java.util.*;

public class HelloWorld {
   public static void main(String[] args) {
                System.out.println("Hello, World!");
        }
}

D:\Teach\!KCC\CP05_2023-03\Assignments\TestSuite\xCheckStyleHeaderTest>java -jar checkstyle.jar -c MinimalConfig.xml HelloWorld.java
Starting audit...
Audit done.
```
